simapp/app.go

```diff
@@ -180,7 +172,7 @@ func NewSimApp(
 		app.cdc, keys[supply.StoreKey], app.AccountKeeper, app.BankKeeper, maccPerms,
 	)
 	stakingKeeper := staking.NewKeeper(
-		app.cdc, keys[staking.StoreKey], app.BankKeeper, app.SupplyKeeper, app.subspaces[staking.ModuleName],
+		appCodec.Staking, keys[staking.StoreKey], app.BankKeeper, app.SupplyKeeper, app.subspaces[staking.ModuleName],
```

Comment:
Okay, I get it now. Would keepers also receive a `Marshaler` instance? Would `AppCodec` include `Marshaler`?

Reply:
> Would keepers also receive a Marshaler instance?

Yes! Unless they need a custom interface like `x/auth` that needs to know how to decode its custom types.